test/Mono.Linker.Tests.Cases/DataFlow/TypeBaseTypeDataFlow.cs

```diff
@@ -0,0 +1,278 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Helpers;
+
+namespace Mono.Linker.Tests.Cases.DataFlow
+{
+	[SkipKeptItemsValidation]
+	public class TypeBaseTypeDataFlow
+	{
+		public static void Main()
+		{
+			TestAllPropagated (typeof (TestType));
+
+			TestPublicConstructorsPropagated (typeof (TestType));
+			TestPublicEventsPropagated (typeof (TestType));
+			TestPublicFieldsPropagated (typeof (TestType));
+			TestPublicMethodsPropagated (typeof (TestType));
+			TestPublicNestedTypesPropagated (typeof (TestType));
+			TestPublicParameterlessConstructorPropagated (typeof (TestType));
+			TestPublicPropertiesPropagated (typeof (TestType));
+
+			TestNonPublicConstructorsAreNotPropagated (typeof (TestType));
+			TestNonPublicEventsAreNotPropagated (typeof (TestType));
+			TestNonPublicFieldsAreNotPropagated (typeof (TestType));
+			TestNonPublicMethodsAreNotPropagated (typeof (TestType));
+			TestNonPublicNestedTypesAreNotPropagated (typeof (TestType));
+			TestNonPublicPropertiesAreNotPropagated (typeof (TestType));
+
+			TestCombinationOfPublicsIsPropagated (typeof (TestType));
+			TestCombinationOfNonPublicsIsNotPropagated (typeof (TestType));
+			TestCombinationOfPublicAndNonPublicsPropagatesPublicOnly (typeof (TestType));
+
+			TestNoAnnotation (typeof (TestType));
+			TestAnnotatedAndUnannotated (typeof (TestType), typeof (TestType), 0);
+			TestNull ();
+
+			Mixed_Derived.Test (typeof (TestType), 0);
```

Comment:
Should this method be called with 1 and 2 as well?

Reply:
The tests actually never run the code, and since we don't do cross-method analysis the actual value here doesn't matter at all. (There's a plan to actually run the tests, but it still would not matter much).